marshalable-derive/src/lib.rs

```diff
@@ -65,25 +94,271 @@ fn derive_tpm_marshal_inner(input: DeriveInput) -> Result<TokenStream> {
             ));
         }
     };
-    let expanded = quote! {
-        #pure_impl
-        // The generated impl.
-        impl Marshalable for #name  {
-            fn try_unmarshal(buffer: &mut tpm2_rs_marshalable::UnmarshalBuf) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<Self> {
-                #unmarsh_text
-            }
+    let expanded = if has_tpm2b_simple || has_tpm2b_struct {
+        quote! {
+            #pure_impl
+        }
+    } else {
+        quote! {
+            #pure_impl
+            // The generated impl.
+            impl Marshalable for #name  {
+                fn try_unmarshal(buffer: &mut tpm2_rs_marshalable::UnmarshalBuf) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<Self> {
+                    #unmarsh_text
+                }
 
-            fn try_marshal(&self, buffer: &mut [u8]) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<usize> {
-                let mut written: usize = 0;
-                #marsh_text;
-                Ok(written)
+                fn try_marshal(&self, buffer: &mut [u8]) -> tpm2_rs_marshalable::exports::errors::TpmRcResult<usize> {
+                    let mut written: usize = 0;
+                    #marsh_text;
+                    Ok(written)
+                }
             }
         }
     };
-
     Ok(expanded)
 }
 
+fn derive_tpm2b_code(
+    tpm2b_outer_struct_name: &Ident,
+    stru: &syn::DataStruct,
+    tpm2b_simple: bool,
+    tpm2b_struct: bool,
+) -> TokenStream {
+    const TPM2B_SIMPLE_ERR_GENERAL: &str = "A Tpm2b struct must contain just to elements, a u16 size and a buffer array when using #[marshalable(tpm2b_simple)]";
```

Comment:
small nit: "two"

Reply:
This is a top level documentation for how to use `tpm2b_simple`, and not for how to use the code generated by `tpm2b_simple`. So I suggest placing it here https://github.com/tpm-rs/tpm-rs/blob/bd995111e44f04bfbd677080d887672feebf60a7/marshalable-derive/src/lib.rs#L12 with an example if possible.

Reply:
Fixed.

Reply:
Added rust comment for the derive, however kept my const string for returning explaining error string.